### ISSUE TITLE
geofence doesn't work on iOS

### ISSUE THREAD

**Issue by wml2017** [shinyorg/shiny#261]
# IF YOU DON'T ANSWER THIS TEMPLATE - THE BOT WILL AUTOMATICALLY CLOSE YOUR ISSUE!

## Please check all of the platforms you are having the issue on (if platform is not listed, it is not supported)

 - [ x] iOS
 - [ ] Android
 - [ ] UWP
 - [ ] watchOS
 - [ ] tvOS
 - [ ] Tizen
 - [ ] .NET Standard

## Feature Library

 - [ ] Core
 - [ ] BluetoothLE
 - [ ] Beacons
 - [ x] Locations
 - [ ] HTTP Transfers
 - [ x] Notifications
 - [ ] Sensors

## Version of Library
1.1.0.1473
## Version of OS(s) listed above with issue
iOS 12.4.6
iOS 13.2.4
## Steps to Reproduce
1. On MacBook Pro, Update Xcode to 11.4 and also update visual studio 2019 mac community to the newest version. Create a new blank Forms App named as Geofencing, only check iOS. After the whole solution is generated, update Xamarin.Forms to the newest version 4.5.0.495 (also update Xamarin.Essential to 1.5.2). Then install Shiny.Notifications 1.1.0.1473 and Shiny.Location 1.1.0.1473

2.  Use following code to replace App.xaml.cs file

using System;
using Xamarin.Forms;
using Xamarin.Forms.Xaml;

using Shiny;
using Shiny.Locations;
using System.Threading.Tasks;
using Shiny.Notifications;
using Microsoft.Extensions.DependencyInjection;

namespace Geofencing
{
    public partial class App : Application
    {
        public App()
        {
            InitializeComponent();

            MainPage = new MainPage();
        }

        protected async override void OnStart()
        {
            var geofences = ShinyHost.Resolve<IGeofenceManager>();
            var notifications = ShinyHost.Resolve<INotificationManager>();

            try
            {                
                var access = await notifications.RequestAccess();
                if (access == AccessState.Available)
                {
                    await geofences.StartMonitoring(new GeofenceRegion(
                        "SF",
                        new Position(34.022564, -118.401250),
                        Distance.FromMeters(200)
                    )
                    {
                        NotifyOnEntry = true,
                        NotifyOnExit = true,
                        SingleUse = false
                    });
                    await App.Current.MainPage.DisplayAlert("Info", "Geofencing is started successfully", "OK");
                }
                else
                {
                    await App.Current.MainPage.DisplayAlert("Warning", "Change settings", "OK");
                }
            }
            catch (Exception ex)
            {
                await App.Current.MainPage.DisplayAlert("Err", ex.Message, "OK");
            }
        }

        protected override void OnSleep()
        {
        }

        protected override void OnResume()
        {
        }
    }

    public class MyGeofenceDelegate : IGeofenceDelegate
    {
        readonly INotificationManager notifications;

        public MyGeofenceDelegate(INotificationManager notifications)
        {
            this.notifications = notifications;
        }

        public async Task OnStatusChanged(GeofenceState newStatus, GeofenceRegion region)
        {
            if (newStatus == GeofenceState.Entered)
            {
                await this.notifications.Send(new Notification
                {
                    Title = "WELCOME!",
                    Message = "It is good to have you back " + region.Identifier
                });
                 //await App.Current.MainPage.DisplayAlert("Info", "Enter SF", "OK");
           }
            else
            {
                await this.notifications.Send(new Notification
                {
                    Title = "GOODBYE!",
                    Message = "You will be missed at " + region.Identifier
                });
                //await App.Current.MainPage.DisplayAlert("Info", "Leave SF", "OK");
            }
        }
    }

    public class SampleStartup : ShinyStartup
    {
        public override void ConfigureServices(IServiceCollection builder)
        {
            builder.UseGeofencing<MyGeofenceDelegate>();
            builder.UseNotifications(true);
        }
    }
}

3. update AppDelegate.cs in Geofencing.iOS project like the following code

using System;
using System.Collections.Generic;
using System.Linq;

using Foundation;
using UIKit;

using Shiny;

namespace Geofencing.iOS
{
    // The UIApplicationDelegate for the application. This class is responsible for launching the 
    // User Interface of the application, as well as listening (and optionally responding) to 
    // application events from iOS.
    [Register("AppDelegate")]
    public partial class AppDelegate : global::Xamarin.Forms.Platform.iOS.FormsApplicationDelegate
    {
        //
        // This method is invoked when the application has loaded and is ready to run. In this 
        // method you should instantiate the window, load the UI into it and then make the window
        // visible.
        //
        // You have 17 seconds to return from this method, or iOS will terminate your application.
        //
        public override bool FinishedLaunching(UIApplication app, NSDictionary options)
        {
            this.ShinyFinishedLaunching(new SampleStartup());
            global::Xamarin.Forms.Forms.Init();
            LoadApplication(new App());

            return base.FinishedLaunching(app, options);
        }
    }
}

4. Add the following into info.plist

<key>NSLocationAlwaysUsageDescription</key>
<string>Your message</string>
<key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
<string>Your message</string>
<key>NSLocationWhenInUseUsageDescription</key>
<string>Your message</string>

<key>UIBackgroundModes</key>
<array>
    <string>location</string>
</array>

Then execute the app

## Expected Behavior
When I close to the given position within 200 meters, a notification, It is good to have you back SF, with WELCOME! title will be sent out.
When I leave the given position more than 200 meters, a notification, You will be missed at SF, with GOODBYE! will be issued by my iPhone.

## Actual Behavior
When I closed to the given position within 200 meters, my iPhone didn't sent any notification.
When I left the given position more than 200 meters, it also didn't sent out a notification.

Because Shiny.Notification also has some problems (please refer my another report at

https://github.com/shinyorg/shiny/issues/260

), I add a message

                await App.Current.MainPage.DisplayAlert("Info", "Enter SF", "OK");

after enter notification and

                await App.Current.MainPage.DisplayAlert("Info", "Leave SF", "OK");

after the leave notification

The two infos were not popped out which means OnStatusChanged wasn't triggered when I entered or left the position's geofence

**Comment by aritchie** [comment]
Geofences are also working just fine for my samples and my apps.  You shouldn't be running XF code in the delegates as there is a good chance xamarin forms isn't ready when things fire in the background

**Comment by wml2017** [comment]
Yes, I just want to know what happens because I wasn't in home when I tested it. Even though I remove 

await App.Current.MainPage.DisplayAlert

it from the simple test app. It didn't work. I also tried your sample some times. It worked yesterday morning after I tried to start it some times. But failed at today's morning with the same path. Also, the Shiny.Location in your sample can be updated to 1.1.0.1487. But my simple app is still on 1.1.0.1473

**Comment by wml2017** [comment]
> Geofences are also working just fine for my samples and my apps. You shouldn't be running XF code in the delegates as there is a good chance xamarin forms isn't ready when things fire in the background

Yes, I just want to know what happens because I had to go out  when I tested it. Even though I remove

await App.Current.MainPage.DisplayAlert

it from the simple test app. It didn't work. I also tried your sample some times. It worked yesterday morning after I tried to start it some times. But failed at today's morning with the same path. Also, the Shiny.Location in your sample can be updated to 1.1.0.1487. But my simple app is still on 1.1.0.1473
